From 68f8d18d40cd65249dfc063de591c0801d44e9e4 Mon Sep 17 00:00:00 2001
From: Tom Pollard <tom.pollard@codethink.co.uk>
Date: Wed, 5 Aug 2026 13:40:18 +0100
Subject: [PATCH] lib/CMakeLists.txt: Include GNUInstallDirs to use
 CMAKE_INSTALL_LIBDIR

This ensures the path for .cmake follows the common pattern to include
the arch, e.g. `usr/lib/x86_64-linux-gnu/cmake/nghttp3`
---
 lib/CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 7dbfde4..1271e05 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -66,15 +66,17 @@ set(nghttp3_SOURCES
   sfparse/sfparse.c
 )
 
+include(CMakePackageConfigHelpers)
+include(GNUInstallDirs)
+
 set(NGHTTP3_GENERATED_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
 set(NGHTTP3_VERSION_CONFIG "${NGHTTP3_GENERATED_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
 set(NGHTTP3_PROJECT_CONFIG "${NGHTTP3_GENERATED_DIR}/${PROJECT_NAME}Config.cmake")
 set(NGHTTP3_TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
-set(NGHTTP3_CONFIG_INSTALL_DIR "lib/cmake/${PROJECT_NAME}")
+set(NGHTTP3_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 set(NGHTTP3_NAMESPACE "${PROJECT_NAME}::")
 set(NGHTTP3_VERSION ${PROJECT_VERSION})
 
-include(CMakePackageConfigHelpers)
 write_basic_package_version_file(
   "${NGHTTP3_VERSION_CONFIG}" VERSION ${NGHTTP3_VERSION} COMPATIBILITY SameMajorVersion
 )
-- 
2.53.0

